src/Microsoft.Extensions.Configuration/ConfigurationRoot.cs

```diff
@@ -24,6 +24,12 @@ public ConfigurationRoot(IList<IConfigurationProvider> providers)
             _providers = providers;
         }
 
+        public string Key => string.Empty;
```

Comment:
I don't think its that unreasonable to change the path format, so basically we'd always have a pathDelimiter so we could distinguish the root value?

":" would be the value of the "" child, while "" would be the value of the root?  Seems reasonable to me :)